routing.go

```diff
@@ -688,9 +674,17 @@ func (dht *IpfsDHT) FindPeer(ctx context.Context, id peer.ID) (_ peer.AddrInfo,
 	}
 
 	// refresh the cpl for this key if we discovered the peer because of the query
-	if ctx.Err() == nil && queries[0].globallyQueriedPeers.Contains(id) {
-		kadID := kb.ConvertPeerID(id)
-		dht.routingTable.ResetCplRefreshedAtForID(kadID, time.Now())
+	if ctx.Err() == nil {
+		discoveredPeerDuringQuery := false
+		for _, p := range lookupRes.peers {
+			if p == id {
+				discoveredPeerDuringQuery = true
+				break
+			}
+		}
+		if discoveredPeerDuringQuery || lookupRes.completed {
+			dht.routingTable.ResetCplRefreshedAtForID(kb.ConvertPeerID(id), time.Now())
```

Comment:
FindPeer gets a special condition to update the routing table if we found the peer during the query since for valid peers we should never complete the query. Maybe this is a bad exception to make though, thoughts?

Reply:
The special condition is that we bother with `discoveredPeerDuringQuery` at all, which is effectively in the current cypress code. We're doing it to be generous, so that successful FindPeers count towards bucket refreshes, but the invariant isn't quite the same (we might not have queried the beta closest peers)

Reply:
@aschmahmann 

I think we discussed this before. Yes, we are being generous here, but as long as we discovered the peer during the query, it means we talked to & formed connections to peers in the same bucket as the peer we are looking for AND so we do not want to bother refreshing that bucket for a while.  

It is not necessary to talk to the beta peers here as we could discover the peer before we talk to the beta peers. Mandating that all disjoint queries complete for us to refresh this bucket when the query has served it's purpose is too restrictive.

Reply:
As agreed. Let's remove bucket refresh on FindPeer, as periodic bucket refreshes cover the need.
We NOTED to update the bucket refresh code to use GetClosestPeer (not FindPeer), to make this work!